### ISSUE TITLE
I want to print multiple dynamic tables to pdf but it shows only the first one 

### ISSUE THREAD

**Issue by Baspa** [simonbengtsson/jsPDF-AutoTable#249]
I have this piece of code:
```
<?php
require 'include/header.html';
require 'php/database.php';

function convertToHoursMins($time, $format = '%02d:%02d')
{
    if ($time < 1) {
        return;
    }
    $hours = floor($time / 60);
    $minutes = ($time % 60);
    return sprintf($format, $hours, $minutes);
}

?>

<script src="js/jspdf.min.js"></script>
<script src="js/jspdf.plugin.autotable.js"></script>

<script type="text/javascript">
    function generate() {

        var doc = new jsPDF('p', 'pt');

        var res = doc.autoTableHtmlToJson(document.getElementById("basic-table"));
        doc.autoTable(res.columns, res.data, {margin: {top: 80}});

        var header = function(data) {
            doc.setFontSize(18);
            doc.setTextColor(40);
            doc.setFontStyle('normal');
            //doc.addImage(headerImgData, 'JPEG', data.settings.margin.left, 20, 50, 50);
            doc.text("Overzicht projecten", data.settings.margin.left, 50);
        };

        var options = {
            beforePageContent: header,
            margin: {
                top: 80
            },
            startY: doc.autoTableEndPosY() + 20
        };

        doc.autoTableHtmlToJson(res.columns, res.data, options);

        doc.save("table.pdf");
    }
</script>

<button onclick="generate()">Generate PDF</button>

<?php
$projectsql = "SELECT * FROM project";
$resultproject = mysqli_query($db,$projectsql);
?>
<?php while($rowproject = mysqli_fetch_array($resultproject)) {
    $id = $rowproject['id'];
    $projecthash = $rowproject['projecthash'];?>
    <table id="basic-table" class="table">
        <thead>
        <tr>
            <th style="width: 40%;"><?php echo $rowproject['projectname']; ?></th>
            <th style="width: 40%;">Taaknaam</th>
            <th style="width: 20%;">Tijd</th>
        </tr>
        </thead>
        <tbody>
        <?php
        $tasksql = "SELECT * FROM task WHERE projecthash = '$projecthash'";
        $resulttask = mysqli_query($db,$tasksql);
        while ($rowtask = mysqli_fetch_array($resulttask)) {
            $taskname = $rowtask['taskname'];

            $sqltotaltime = "SELECT SUM(activityduration) AS value FROM activity WHERE taskconnection = '$taskname'";
            $resulttotaltime = mysqli_query($db, $sqltotaltime);
            $rowtotaltime = mysqli_fetch_array($resulttotaltime);
            $totaltime = $rowtotaltime['value'];
            $total = convertToHoursMins($totaltime, '%02d uur %02d minuten');
            ?>
            <tr>
                <td></td>
                <td><?php echo $taskname ?></td>
                <td><?php echo $total ?></td>
            </tr>
        <?php } ?>
        </tbody>
    </table>
<?php } ?>
</div>
</body>
</html>
```

Which shows me this page:
![1](https://cloud.githubusercontent.com/assets/10845460/24702953/5f4d2bba-1a01-11e7-8594-d43f6ccfc99e.png)

But only outputs this:
![2](https://cloud.githubusercontent.com/assets/10845460/24702959/6502d438-1a01-11e7-8421-7872841627ae.png)

Someone who can explain to me what I'm doing wrong?

**Comment by simonbengtsson** [comment]
In your second call to autoTableHtmlToJson the input should be a HTML element.

**Comment by RajiLaksh** [comment]
I need help in downloading pdf for  dynamic table. Right now I'm able to fetching only first row. Please help me to print with all the rows.

**JS code**

   function generate() {
    	  let doc = new jsPDF('p', 'pt');
			
    	  
    	   	  
    	  
    	  const header = function(data) {
    	    doc.setFontSize(20);
    	    doc.setTextColor(200, 0, 255);
    	    doc.setFontStyle('normal');
    	    doc.text('Report', data.settings.margin.left + 35, 60);
    	    
    	    let today = new Date();
    	    let dd = today.getDate();
    	    //let month = monthNames[today.getMonth()]; //January is 0!
    	    let yyyy = today.getFullYear();
			
    	    var monthNames = ["January", "February", "March", "April", "May", "June",
    	                      "July", "August", "September", "October", "November", "December"
    	                    ];

    	                    
							let month =  monthNames[today.getMonth()]
    	                    console.log("The current month is " + monthNames[today.getMonth()]);
    	    if(dd<10) {
    	        dd = '0'+dd
    	    } 

    	    

    	    today = dd + ' ' + month + ' ' + yyyy;
    	    var newdat = today;
    	    doc.text(107,68,today);
    	    
    	  };

    	  const totalPagesExp = '{total_pages_count_string}';
    	  const footer = function(data) {
    	    let str = 'Testing ' + data.pageCount;
    	    // Total page number plugin only available in jspdf v1.0+
    	    if (typeof doc.putTotalPages === 'function') {
    	      str = str + ' - ' + totalPagesExp+' - ';
    	      console.log('test');
    	    }
    	    doc.text(str, data.settings.margin.left, doc.internal.pageSize.height - 30);
    	  };

    	  const options = {
    	    beforePageContent: header,
    	    afterPageContent: footer,
    	    margin: {
    	      top: 200
    	    }
    	  };

    	  var elem = document.getElementById('table');
    	  var data = doc.autoTableHtmlToJson(elem);
    	  doc.autoTable(data.columns, data.rows, options);

    	  if (typeof doc.putTotalPages === 'function') {
    	    doc.putTotalPages(totalPagesExp);
    	  }
    	  
    	  doc.save('tabled.pdf');
    	}
